pkg/condition/condition.go

```diff
@@ -110,14 +133,6 @@ func (c *EvaluableCondition) CastContextToTypedParameters(contextMap map[string]
 		converted[key] = convertedParam
 	}
 
-	// validate against extraneous parameters
-	for key := range contextMap {
```

Comment:
Dropped this here since it is strictly used for validations when applied in the `Write` API. When casting context parameters to typed params in query evaluations we don't want to apply this behavior.